easybuild/toolchains/compiler/gcc.py

```diff
@@ -66,15 +69,17 @@ class Gcc(Compiler):
 
     # used when 'optarch' toolchain option is enabled (and --optarch is not specified)
     COMPILER_OPTIMAL_ARCHITECTURE_OPTION = {
-        systemtools.AMD : 'march=native',
-        systemtools.INTEL : 'march=native',
-        systemtools.POWER: 'mcpu=native',  # no support for march=native on POWER
+        systemtools.AARCH32: 'mcpu=native', # implies -march=native and -mtune=native
+        systemtools.AARCH64: 'mcpu=native', # since GCC 6; implies -march=native and -mtune=native
+        systemtools.POWER: 'mcpu=native',   # no support for -march on POWER; implies -mtune=native
+        systemtools.X86_64: 'march=native', # implies -mtune=native
     }
     # used with --optarch=GENERIC
     COMPILER_GENERIC_OPTION = {
-        systemtools.AMD : 'march=x86-64 -mtune=generic',
-        systemtools.INTEL : 'march=x86-64 -mtune=generic',
-        systemtools.POWER: 'mcpu=generic-arch',  # no support for -march on POWER
+        systemtools.AARCH32: 'mcpu=generic-armv7', # implies -march=armv7 and -mtune=generic-armv7
+        systemtools.AARCH64: 'mcpu=generic',       # implies -march=armv8-a and -mtune=generic
+        systemtools.POWER: 'mcpu=generic-arch',    # no support for -march on POWER
```

Comment:
@pescobar: it seems like you added this while implementing support for `--optarch=GENERIC`? cfr. 7f44bb6

Any pointers to where you got this from?